rmf_traffic/include/rmf_traffic/schedule/Viewer.hpp

```diff
@@ -80,25 +81,33 @@ class Viewer
 
   /// Query this Viewer to get a View of the Trajectories inside of it that
   /// match the Query parameters.
-  View query(const Query& parameters) const;
+  virtual View query(const Query& parameters) const = 0;
 
-  /// Get the oldest version number inside this Database.
-  Version oldest_version() const;
+  // TODO(MXG): Consider providing an iterator-style API to view participant IDs
+  // and participant descriptions.
 
-  /// Get the latest version number of this Database.
-  Version latest_version() const;
+  /// Get the set of active participant IDs.
+  virtual const std::unordered_set<ParticipantId>& participant_ids() const = 0;
+
+  /// Get the information of the specified participant if it is available.
+  /// If a participant with the specified ID is not registered with the
+  /// schedule, then this will return a nullopt.
```

Comment:
```suggestion
  /// schedule, then this will return a nullptr.
```
or should the return type be rmf_utils::optional<ParticipantDescription> instead?